pkgs/development/python-modules/courlan/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, langcodes
+, pytestCheckHook
+, tld
+, urllib3
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "courlan";
+  version = "0.9.5";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-ONw1suO/H11RbQDVGsEuveVD40F8a+b2oic8D8W1s1M=";
+  };
+
+  propagatedBuildInputs = [
+    langcodes
+    tld
+    urllib3
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  # disable tests that require an internet connection
+  disabledTests = [
+    "test_urlcheck"
+  ];
+
+  # nixify path to the courlan binary in the test suite
+  postPatch = ''
+    substituteInPlace tests/unit_tests.py \
+      --replace '"courlan --help"' "'$out/bin/courlan --help'" \
+      --replace 'courlan_bin = "courlan"' "courlan_bin = '$out/bin/courlan'"
```

Comment:
If we quote the $out in single quotes, isn't that literal written into the file and never substituted to the actual path?

Reply:
Ah, you are right. I just didn't notice because the underlying test is able to access $out itself. Probably still better to fix though.

Reply:
Just FYI to avoid the quoting hell: there is ${placeholder "out"}

Reply:
That is very handy, thanks for the tip! :)